### ISSUE TITLE
Send push if item.title start with "!"

### ISSUE THREAD

**Issue by vdias38** [meumobi/sitebuilder#660]
No matter if category is push enabled or not, when item.title starts with "!" it should send a push when item is created

**Comment by danconte72** [comment]
should ! be trimmed from title before creation?
"!1T19" => "1T19"

**Comment by vdias38** [comment]
yes the "!" should be trimmed

**Comment by danconte72** [comment]
### Actual Behavior
- Create an item with title starting with "!" on a not enabled push category
```
[2019-05-23 18:25:40] sitebuilder.INFO: "item created" {"item_id":"5ce6e5a410f0ed6e1b8b4567","site_id":3,"category_id":14,"downloaded_images":0,"failed_images":0,"component":"create_item"}
```
### Expected Behavior
- Create an item with title starting with "!" on a not enabled push category
```
[2019-05-23 18:05:04] sitebuilder.INFO: "item created" {"item_id":"5ce6e0d010f0edf01c8b4567","site_id":3,"category_id":14,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-23 18:05:04] sitebuilder.INFO: "job created" {"job_id":"5ce6e0d010f0edf01c8b4568","type":"push_notification","params":{"item_id":"5ce6e0d010f0edf01c8b4567","category_id":14},"component":"worker_manager"}
```
### How it works
- Push Notification is only sent when Item category is push enabled
### What should be updated
- Update Items Model schema to add push flag (boolean, default, null)
- On CreateItem event
  -  Tests If title starts with "!". If true 
     1. flag item to send push
     2. trim ! from title
     3. save Item
- Update Push Notification Check to consider either item push flag==true OR Item category push==true to call PushNotification service
- Push Notification Check must be updated on CreateItem and PublishItem service

### Implementation flow
- Update Items model
- Update CreateItem event
- Update PushNotificationCheck on CreateItem service
- Update PushNotificationCheck on PublishItem service

**Comment by vdias38** [comment]
- I don't understand the Actual behavior and Expected behavior sections, the purpose of issue is not to change logs.
- how it works seems to correspond to actual behavior section.
- Can you explain how you pretend to update the "Push Notification Check"? current check is `if ($sendPush && $item->parent()->notification)` how you'll integrate "item push flag==true OR Item category push==true" is the response expected on this comment.

I recommend to create item.push flag on item created, item.push =category.push OR item.title startsWith(!). Should streamlines what you call "Push Notification Check". 
And should consider following cases:
- item created from Feed
  - send push if item.push and extension.isLowPriority
- item created fom UI
  - send push if item.push

Check #663 before to implement this issue.

**Comment by danconte72** [comment]
### Tests 
1. Create Item Title = !teste on UI , category-push=false// SHOULD CHECK/SEND PUSH
```
[2019-05-27 18:39:08] sitebuilder.INFO: "item created" {"item_id":"5cec2ecc10f0edb21f8b4569","site_id":3,"is_published":true,"category_id":15,"push":true,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-27 18:39:08] sitebuilder.INFO: "job created" {"job_id":"5cec2ecc10f0edb21f8b456b","type":"push_notification","params":{"item_id":"5cec2ecc10f0edb21f8b4569","category_id":15},"component":"worker_manager"}
```
2. Create Item Title = teste on UI , category-push=true// SHOULD CHECK/SEND PUSH
```
[2019-05-27 18:50:37] sitebuilder.INFO: "item created" {"item_id":"5cec317d10f0edaf1f8b4569","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-27 18:50:37] sitebuilder.INFO: "job created" {"job_id":"5cec317d10f0edaf1f8b456b","type":"push_notification","params":{"item_id":"5cec317d10f0edaf1f8b4569","category_id":15},"component":"worker_manager"}
```
3. Create Item Title = !teste on UI , category-push=true// SHOULD CHECK/SEND PUSH
```
[2019-05-27 18:50:47] sitebuilder.INFO: "item created" {"item_id":"5cec318710f0edb11f8b456c","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
[2019-05-27 18:50:47] sitebuilder.INFO: "job created" {"job_id":"5cec318710f0edb11f8b456e","type":"push_notification","params":{"item_id":"5cec318710f0edb11f8b456c","category_id":15},"component":"worker_manager"}
```

4.  Create Item Title = teste on UI , category-push=false// SHOULD NOT CHECK/NOT SEND PUSH
```[2019-05-27 18:53:51] sitebuilder.INFO: "item created" {"item_id":"5cec323f10f0edae1f8b4567","site_id":3,"is_published":true,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
```

5. Create Item Title = !teste on UI , category-push=false, published date on future
-  SHOULD NOT CHECK/NOT SEND PUSH WHEN CREATED
```
[2019-05-27 18:44:39] sitebuilder.INFO: "item created" {"item_id":"5cec301710f0edc11f8b4567","site_id":3,"is_published":false,"category_id":15,"downloaded_images":0,"failed_images":0,"component":"create_item"}
```
- SHOULD CHECK/SEND PUSH WHEN PUBLISHED
```
[2019-05-27 18:45:25] sitebuilder.INFO: item published {"item_id":"5cec2ff110f0edb91f8b4567","is_published":true,"component":"publish_items"} []
[2019-05-27 18:45:25] sitebuilder.INFO: job created {"job_id":"5cec304510f0edac208b4567","type":"push_notification","params":{"item_id":"5cec2ff110f0edb91f8b4567","category_id":15},"component":"worker_manager"} []
```
